include/net/ieee802154_radio.h

```diff
@@ -209,10 +209,8 @@ struct ieee802154_radio_api {
 			 enum ieee802154_config_type type,
 			 const struct ieee802154_config *config);
 
-#ifdef CONFIG_NET_L2_IEEE802154_SUB_GHZ
```

Comment:
Why do we remove this?

Reply:
This device supports `CONFIG_NET_L2_IEEE802154` and `CONFIG_NET_L2_IEEE802154_SUB_GHZ` simultaneously. Also, if `CONFIG_IEEE802154_RAW_MODE` is set, then neither `CONFIG_NET_L2_IEEE802154` or `CONFIG_NET_L2_IEEE802154_SUB_GHZ` are enabled, which results in 2.4 GHz radios being able to change their channels but not SubGHz radios.

Reply:
fine by me